SortingAlgorithms/Scenes/AlgorithmDetail/AlgorithmDetailViewController.swift

```diff
@@ -83,16 +98,29 @@ final class AlgorithmDetailViewController: UIViewController {
     @objc private func closeButtonDidTouchUpInside() {
         presenter.closeButtonTriggered()
     }
+    
+    @objc private func restartButtonDidTouchUpInside() {
+        presenter.restart()
```

Comment:
Aqui seria interessante não dar inteligência para a View, no quase quando algo for delegado para a Presenter, delegar de modo que a presenter receba o "acontecimento" da view e nâo uma ordem ou ação. 

Sugestão:` presenter.restartButtonTriggered()`